relay-filter/src/config.rs

```diff
@@ -328,15 +482,215 @@ impl FiltersConfig {
     }
 }
 
+/// Configuration of generic filters combined from project and global config.
+///
+/// See [`DynamicFiltersConfigIter`] for details on how to iterate easily
+/// through the applicable filters.
+#[derive(Copy, Clone, Debug)]
+pub(crate) struct DynamicFiltersConfig<'a> {
```

Comment:
Doesn't have to be exactly this name, but `Generic` should be part of it IMO.

```suggestion
pub(crate) struct DynamicGenericFiltersConfig<'a> {
```